Bottom: b2f7b658fae5c5c1a2c58c096c1807cee9423a00
Top:    5199b5ada6f1b889fa1fc1adcf706f7763463556
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-25 12:30:40 -0700

Refresh of rs-with-acm

---

diff --git a/src/acm.c b/src/acm.c
index d59efcf..e9733c6 100755
--- a/src/acm.c
+++ b/src/acm.c
@@ -99,24 +99,18 @@ void ucma_acm_init(void)
 	ucma_set_server_port();
 	sock = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
 	if (sock < 0)
-		goto err1;
+		goto out;
 
 	memset(&addr, 0, sizeof addr);
 	addr.sin_family = AF_INET;
 	addr.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
 	addr.sin_port = htons(server_port);
 	ret = connect(sock, (struct sockaddr *) &addr, sizeof(addr));
-	if (ret)
-		goto err2;
-
-	init = 1;
-	pthread_mutex_unlock(&acm_lock);
-	return;
-
-err2:
-	close(sock);
-	sock = -1;
-err1:
+	if (ret) {
+		close(sock);
+		sock = -1;
+	}
+out:
 	init = 1;
 	pthread_mutex_unlock(&acm_lock);
 }
